### PR TITLE
fix(dropdown): fix a11y issues for dropdown

### LINKED ISSUE #13110: [a11y]: Dropdown has critical violation when the dropdown list is open: None of the descendent elements with "option" role is tabbable
### Package

carbon-components, carbon-components-react

### Browser

Firefox

### Operating System

MacOS

### Package version

7.57.4

### React version

17.0.2

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker: Latest Deployment

### Assistive technology

_No response_

### Description

When the dropdown list is open, IBM Equal Access Accessibility Checker gives a violation as shown in the attached screenshot. This happens in both V10 and V11 versions of carbon components.

image
image

### WCAG 2.1 Violation

_No response_

### Reproduction/example

https://carbondesignsystem.com/components/dropdown/usage/

### Steps to reproduce

1. Go to the carbon design system website.
2. Select the component Dropdown.
3. Open the dropdown and run the Accessibility checker tool. The violation will be reported by the tool

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

### PR DESCRIPTION
**Pull request by aledavila**
Closes #13110

Fixes a11y issues with dropdown.

#### Changelog

**New**

- Adds `role=combobox` to dropdown trigger.

#### Testing / Reviewing

Make sure a11y violations are fixed.

### CODE REVIEW

**Comment by tay1orjones** [comment]
@aledavila It looks like the modal story that includes a dropdown is now failing the automated accessibility testing.

File: packages/react/src/components/Dropdown/Dropdown.tsx [hunk 1]
```diff
@@ -386,6 +386,8 @@ const Dropdown = React.forwardRef(
           )}
           <button
             type="button"
+            role="combobox" // eslint-disable-line jsx-a11y/role-has-required-aria-props
+            aria-controls={getMenuProps().id}
```
Comment (francinelucca) [inline packages/react/src/components/Dropdown/Dropdown.tsx:391]: could we circunvent adding the eslint-disable by addind the required aria-expanded attr?  or would that interfere with something? 
```suggestion
            role="combobox"
            aria-expanded={isOpen}
            aria-controls={getMenuProps().id}
```

**Review by francinelucca — COMMENTED** [review]
Confirmed no longer see a11y violation and dropdown still works and looks as expected 🚀
Just one question on eslint ruling

File: packages/react/src/components/Dropdown/Dropdown.tsx (hunk 1 above)
Comment (aledavila) [reply packages/react/src/components/Dropdown/Dropdown.tsx:391]: It has it under the hood. Its just that its coming from the downshift props through `{...toggleButtonProps}` so eslint thinks it doesn't have it. 
Screen Shot 2023-03-30 at 5 40 43 PM

Comment (francinelucca) [reply packages/react/src/components/Dropdown/Dropdown.tsx:391]: nice! could we do `aria-expanded={toggleButtonProps['aria-expanded']}` then? or alternatively add a comment explaining that that attribute is already added by toggleButtonProps? I just want it to be understood that we're not choosing to ignore an a11y rule 😅

Comment (aledavila) [reply packages/react/src/components/Dropdown/Dropdown.tsx:391]: True I'll add a better comment instead cause idk if adding double aria-expanded would mess it up

**Review by tw15egan — APPROVED** [review]

File: packages/react/src/components/Dropdown/Dropdown.tsx (hunk 1 above)
Comment (aledavila) [reply packages/react/src/components/Dropdown/Dropdown.tsx:391]: @francinelucca added it

**Review by francinelucca — APPROVED** [review]
🚀

**Pull request merged**